SimTransport/TotemRPProtonTransportParametrization/src/LHCOpticsApproximator.cc

```diff
@@ -0,0 +1,1212 @@
+
+  std::string x_out_lab = data_prefix + "_x_out";
+  std::string theta_x_out_lab = data_prefix + "_theta_x_out";
+  std::string y_out_lab = data_prefix + "_y_out";
+  std::string theta_y_out_lab = data_prefix + "_theta_y_out";
+  std::string ksi_out_lab = data_prefix + "_ksi_out";
+  std::string s_out_lab = data_prefix + "_s_out";
+  std::string valid_out_lab = data_prefix + "_valid_out";
+
+  //disable not needed branches to speed up the readin
+  inp_tree->SetBranchStatus("*",false);  //disable all branches
+  inp_tree->SetBranchStatus(x_in_lab.c_str(),true);
+  inp_tree->SetBranchStatus(theta_x_in_lab.c_str(),true);
+  inp_tree->SetBranchStatus(y_in_lab.c_str(),true);
+  inp_tree->SetBranchStatus(theta_y_in_lab.c_str(),true);
+  inp_tree->SetBranchStatus(ksi_in_lab.c_str(),true);
+  inp_tree->SetBranchStatus(x_out_lab.c_str(),true);
+  inp_tree->SetBranchStatus(theta_x_out_lab.c_str(),true);
+  inp_tree->SetBranchStatus(y_out_lab.c_str(),true);
+  inp_tree->SetBranchStatus(theta_y_out_lab.c_str(),true);
+  inp_tree->SetBranchStatus(ksi_out_lab.c_str(),true);
+  inp_tree->SetBranchStatus(valid_out_lab.c_str(),true);
+
+  //set input data adresses
+  inp_tree->SetBranchAddress(x_in_lab.c_str(), &(in_var[0]) );
+  inp_tree->SetBranchAddress(theta_x_in_lab.c_str(), &(in_var[1]) );
+  inp_tree->SetBranchAddress(y_in_lab.c_str(), &(in_var[2]) );
+  inp_tree->SetBranchAddress(theta_y_in_lab.c_str(), &(in_var[3]) );
+  inp_tree->SetBranchAddress(ksi_in_lab.c_str(), &(in_var[4]) );
+  inp_tree->SetBranchAddress(s_in_lab.c_str(), &(in_var[5]) );
+
+  //set output data adresses
+  inp_tree->SetBranchAddress(x_out_lab.c_str(), &(out_var[0]) );
+  inp_tree->SetBranchAddress(theta_x_out_lab.c_str(), &(out_var[1]) );
+  inp_tree->SetBranchAddress(y_out_lab.c_str(), &(out_var[2]) );
+  inp_tree->SetBranchAddress(theta_y_out_lab.c_str(), &(out_var[3]) );
+  inp_tree->SetBranchAddress(ksi_out_lab.c_str(), &(out_var[4]) );
+  inp_tree->SetBranchAddress(s_out_lab.c_str(), &(out_var[5]) );
+  inp_tree->SetBranchAddress(valid_out_lab.c_str(), &(out_var[6]) );
+
+  Long64_t entries = inp_tree->GetEntries();
+  if(entries>0)
+  {
+    inp_tree->SetBranchStatus(s_in_lab.c_str(),true);
+    inp_tree->SetBranchStatus(s_out_lab.c_str(),true);
+    inp_tree->GetEntry(0);
+    s_begin_ = in_var[5];
+    s_end_ = out_var[5];
+    inp_tree->SetBranchStatus(s_in_lab.c_str(),false);
+    inp_tree->SetBranchStatus(s_out_lab.c_str(),false);
+  }
+
+  //set input and output variables for fitting
+  for(Long64_t i=0; i<entries; ++i)
+  {
+    inp_tree->GetEntry(i);
+    if(out_var[6] != 0)  //if out data valid
+    {
+      x_parametrisation.AddRow(in_var, out_var[0], 0);
+      theta_x_parametrisation.AddRow(in_var, out_var[1], 0);
+      y_parametrisation.AddRow(in_var, out_var[2], 0);
+      theta_y_parametrisation.AddRow(in_var, out_var[3], 0);
+    }
+  }
+
+  std::cout<<"Optical functions parametrizations from "<<s_begin_<<" to "<<s_end_<<std::endl;
```

Comment:
@mundim @civanch also here could we replace std::cout with the MessageLogger?

Reply:
This code comes from TOTEM and we would like to keep it as close to the original as possible. New developments will be done in such a way it works both in the TOTEM alone and CMS framework, so, we would strongly like to avoid including anything from CMS since the more we change it, the more difficult it will be to include future developments.
So, would it be too bad to keep std::cout instead of using MessageLogger? Thanks

Reply:
the other comment is implemented. I'm just waiting for the reply on the request above to push the new code to avoid restarting all the change and then have to push again.

Reply:
Ok. I'll proceed with the needed changes. Thanks.